### Brady kids' ages
Greg was 14, Marcia was 12, Peter was 11, Jan was 10, Bobby was 8, and Cindy was 6 when they started playing the Brady kids on The Brady Bunch. Cousin Oliver was 8.

In [29]:
import pandas as pd
import numpy as np
import matplotlib as plt
names = ['Greg', 'Marcia', 'Peter', 'Jan', 'Bobby','Cindy','Oliver']
brady = pd.DataFrame(index=names)
brady['ages']=[14,12,11,10,8,6,8]
b_mean = brady.mean()
b_median = brady.median()
b_mode = brady.mode()
print(brady.describe())
print('Variance:',brady['ages'].var())
print('Mean:', b_mean, '\n',
     'Median:', b_median, '\n',
     'Mode:', b_mode)
print('Standard Error:', np.std(brady['ages'] ,ddof=1) / np.sqrt(len(brady['ages'])))


            ages
count   7.000000
mean    9.857143
std     2.734262
min     6.000000
25%     8.000000
50%    10.000000
75%    11.500000
max    14.000000
Variance: 7.476190476190475
Mean: ages    9.857143
dtype: float64 
 Median: ages    10.0
dtype: float64 
 Mode:    ages
0     8
Standard Error: 1.0334540197243192


### Measuring Central Tendency and Variance
For central tendency I would choose Jan because her age (10) is closest to the mean (9.86).

For variance I would choose Bobby or Oliver (both 8) because their ages' are closest to the variance (7.47).

### Cindy's Birthday
If Cindy had a birthday (from 6 to 7), She would be the chosen for proxomity to variance.

In [32]:
import pandas as pd
import numpy as np
import matplotlib as plt
names = ['Greg', 'Marcia', 'Peter', 'Jan', 'Bobby','Cindy','Oliver']
brady = pd.DataFrame(index=names)
brady['ages']=[14,12,11,10,8,7,8]
b_mean = brady.mean()
b_median = brady.median()
b_mode = brady.mode()
print(brady.describe())
print('Variance:',brady['ages'].var())
print('Mean:', b_mean, '\n',
     'Median:', b_median, '\n',
     'Mode:', b_mode)
print('Standard Error:', np.std(brady['ages'] ,ddof=1) / np.sqrt(len(brady['ages'])))

            ages
count   7.000000
mean   10.000000
std     2.516611
min     7.000000
25%     8.000000
50%    10.000000
75%    11.500000
max    14.000000
Variance: 6.333333333333333
Mean: ages    10.0
dtype: float64 
 Median: ages    10.0
dtype: float64 
 Mode:    ages
0     8
Standard Error: 0.9511897312113418


### Replacing Oliver
Replacing Oliver(8) with Jessica(1) shifts the cental tendency down to 9 from 10 and moves the variance to 18.

In [31]:
import pandas as pd
import numpy as np
import matplotlib as plt
names = ['Greg', 'Marcia', 'Peter', 'Jan', 'Bobby','Cindy','Jessica']
brady = pd.DataFrame(index=names)
brady['ages']=[14,12,11,10,8,6,1]
b_mean = brady.mean()
b_median = brady.median()
b_mode = brady.mode()
print(brady.describe())
print('Variance:',brady['ages'].var())
print('Mean:', b_mean, '\n',
     'Median:', b_median, '\n',
     'Mode:', b_mode)
print('Standard Error:', np.std(brady['ages'] ,ddof=1) / np.sqrt(len(brady['ages'])))

            ages
count   7.000000
mean    8.857143
std     4.336995
min     1.000000
25%     7.000000
50%    10.000000
75%    11.500000
max    14.000000
Variance: 18.80952380952381
Mean: ages    8.857143
dtype: float64 
 Median: ages    10.0
dtype: float64 
 Mode:    ages
0     1
1     6
2     8
3    10
4    11
5    12
6    14
Standard Error: 1.6392299502912862


## 50th anniversary
On the 50th anniversary of The Brady Bunch, four different magazines asked their readers whether they were fans of the show. The answers were: TV Guide 20% fans Entertainment Weekly 23% fans Pop Culture Today 17% fans SciPhi Phanatic 5% fans

In [26]:
mags = ['TV Guide', 'Entertainment Weekly', 'Pop Culture Today', 'SciPhi Phanatic']
mags = pd.DataFrame(index=mags)
mags['percentages']=[.2,.23,.17,.05]
print('Variance:',mags['percentages'].var())
mags.describe()

Variance: 0.0062250000000000005


percentages
count     4.000000
mean      0.162500
std       0.078899
min       0.050000
25%       0.140000
50%       0.185000
75%       0.207500
max       0.230000

Based on the information it couldy be fair to say that 16% of Americans would have been fans of the Brady 50th annivesary event.

### Without the SciPhi magazine (...with help from hints)
Without the data from the SciPhi information the percentage of fans moves to 20% from 16%.

In [27]:
mags_no_sciphi = ['TV Guide', 'Entertainment Weekly', 'Pop Culture Today']
mags_no_sciphi = pd.DataFrame(index=mags_no_sciphi)
mags_no_sciphi['percentages-sciphi']=[.2,.23,.17]
mags_no_sciphi.describe()

percentages-sciphi
count               3.000
mean                0.200
std                 0.030
min                 0.170
25%                 0.185
50%                 0.200
75%                 0.215
max                 0.230